### ISSUE TITLE
Appending to wrong element

### ISSUE THREAD

**Issue by christianmagill** [scotteh/php-dom-wrapper#18]
I'm running into an issue with items being appended to the wrong element.

I am trying to wrap the below html into a fieldset and change the label to a legend. 

The code being used

```
$doc = new Document();
$doc->setLibxmlOptions(LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
$doc->html( "<label class='label'>Test Label</label><div class='fields'><input/></div>" );

$label = $doc->find('.label')->first();
$fieldset = $doc->createElement('fieldset');
$legend = $doc->createElement( 'legend' );

$legend->append($label->contents());
$label->remove();

$fields = $doc->find('.fields');
$fields->detach();

$fieldset->append($legend);
$fieldset->append($fields);

$doc->append($fieldset);

echo htmlentities($doc->saveHTML($doc) );

die();
```

The resulting html:

`<label class="label">Test Label</label><fieldset><legend>Test Label<div class="fields"><input></div> </legend></fieldset>`

But $fields was appended to $fieldset, so why is it inside of the legend?

**Comment by scotteh** [comment]
Two issues here:

1. Specifying 'LIBXML_HTML_NOIMPLIED' appears to only like dealing with 1 root element and will re-arrange any subsequent elements inside the first. Committed a fix for this. This fix will only work when setting html via the `->html()` method.
2. The detach method returns a new NodeList. You'll need to do `$fields = $fields->detach();`.

**Comment by scotteh** [comment]
Spun the fix out to work with loadHTML() in c7ab15c.
